Automattic/wp-calypso
# Site-level billing: Add receipt view

#### Changes proposed in this Pull Request

This adds the site-level version of the receipt page at the url `/purchases/billing-history/:siteSlug/:receiptId` building on the site-level billing-history section added in https://github.com/Automattic/wp-calypso/pull/46175.

Fixes https://github.com/Automattic/wp-calypso/issues/46203

Main project thread: pbOQVh-sc-p2

#### Screenshots

![Screen Shot 2020-10-07 at 4 33 42 PM](https://user-images.githubusercontent.com/2036909/95384896-41285180-08bb-11eb-93c0-0b09e4f9194c.png)

#### To do

- [x] Add route
- [x] Add page
- [x] Add link to receipt page from history page
- [x] Add Back button and header
- [x] Add redirect for invalid receipts
- [x] Add redirect for invalid site
- [x] Add analytics calls (eg: Print button)

#### Testing instructions

First, because this modifies the account-level receipt page (for easier re-use), we should test that to be sure it is not broken.

- Visit `http://calypso.localhost:3000/me/purchases/billing`.
- Click on "View receipt" for any purchase.
- Verify that you see the correct receipt.
- Verify that the sidebar remains at the account-level.
- Click the "Back" link in the header and verify that you're taken back to the account-level billing history page.

Then we can test the site-level receipt page.

- Visit `http://calypso.localhost:3000/purchases/billing-history/:siteSlug`.
- Click on "View receipt" for any purchase.
- Verify that you see the correct receipt.
- Verify that the sidebar remains at the site-level.
- Click the "Back" link in the header and verify that you're taken back to the site-level billing history page.
- Visit the receipt page again.
- Verify that the "Print" button causes the browser to open a print dialog.
- Manually visit the URL of a receipt that doesn't exist and verify that you are redirected to the site-level billing history page for that site.
- Manually visit the URL of a receipt that _does exist_ but one that exists for a different site than the slug in the URL and verify that you are redirected to the site-level billing history page for the site in the URL.
